pkg/ccl/backupccl/restore_online.go

```diff
@@ -312,33 +312,52 @@ func (r *restoreResumer) maybeCalculateTotalDownloadSpans(
 }
 
 func (r *restoreResumer) sendDownloadWorker(
-	execCtx sql.JobExecContext, downloadSpansCh chan roachpb.Spans,
+	execCtx sql.JobExecContext, spans roachpb.Spans, completionPoller chan struct{},
 ) func(context.Context) error {
 	return func(ctx context.Context) error {
 		ctx, tsp := tracing.ChildSpan(ctx, "backupccl.sendDownloadWorker")
 		defer tsp.Finish()
-		for sp := range downloadSpansCh {
-			log.VInfof(ctx, 1, "sending download request for %d spans", len(sp))
+
+		for rt := retry.StartWithCtx(
+			ctx, retry.Options{InitialBackoff: time.Millisecond * 100, MaxBackoff: time.Second * 10},
+		); ; rt.Next() {
+			if err := ctx.Err(); err != nil {
+				return err
+			}
+			log.Infof(ctx, "sending download request for %d spans", len(spans))
 			var resp *serverpb.DownloadSpanResponse
 			var err error
 			if resp, err = execCtx.ExecCfg().TenantStatusServer.DownloadSpan(ctx, &serverpb.DownloadSpanRequest{
-				Spans: sp,
+				Spans: spans,
 			}); err != nil {
 				return err
 			}
-			if len(resp.ErrorsByNodeID) > 0 {
-				return errors.Newf("failed to download spans on all nodes: %v", resp.ErrorsByNodeID)
+			log.Infof(ctx, "finished sending download requests for %d spans, %d errors", len(spans), len(resp.ErrorsByNodeID))
+			for n, err := range resp.ErrorsByNodeID {
+				return errors.Newf("failed to download spans on %d nodes; n%d returned %v", len(resp.ErrorsByNodeID), n, err)
+			}
+
+			// Wait for the completion poller to signal that it has checked our work.
+			select {
+			case _, ok := <-completionPoller:
+				if !ok {
+					return nil
+				}
```

Comment:
add a log line to indicate that we're running it back?

Reply:
no need, we'll just get the `sending download request` line again, right?

Reply:
Ah sure.